File: address-geocode-us-python-examples/get_reverse_location_soap_sdk_example.py
import sys
import os

sys.path.insert(0, os.path.abspath("../address-geocode-us-python/SOAP"))

from get_reverse_location_soap import GetReverseLocationSoap

def get_reverse_location_soap_sdk_go(is_live: bool, license_key: str) -> None:
    print("\r\n--------------------------------------------------")
    print("Address Geocode US - GetReverseLocation - SOAP SDK")
    print("--------------------------------------------------")

    latitude = "34.419120"
    longitude = "-119.703421"
    timeout_seconds = 15

    print("\r\n* Input *\r\n")
    print(f"Latitude        : {latitude}")
    print(f"Longitude       : {longitude}")
    print(f"License Key     : {license_key}")
    print(f"Is Live         : {is_live}")
    print(f"Timeout Seconds : {timeout_seconds}")

    try:
        service = GetReverseLocationSoap(latitude,longitude,license_key,is_live,timeout_seconds)
        response = service.get_reverse_location()

        if response and not hasattr(response, "Error"):
            print("\r\n* Reverse Location Info *\r\n")
            print(f"Address    : {response.Address}")
            print(f"City       : {response.City}")
            print(f"County     : {response.County}")
            print(f"State      : {response.State}")
            print(f"Zip        : {response.Zip}")
            print(f"PostalCode : {getattr(response, 'PostalCode', '')}")
        elif hasattr(response, "Error") and response.Error:
            print("\n* Error *\n")
            print(f"Error Description: {response.Error.Desc}")
            print(f"Error Number     : {response.Error.Number}")
            print(f"Error Location   : {response.Error.Location}")
        else:
            print("\nNo location info found.")
    except Exception as e:
        print("\n* Exception *\n")
        print(f"Exception Message: {str(e)}")

File: address-geocode-us-nodejs-examples/get_reverse_location_soap_sdk_example.js
import { GetReverseLocationSoap } from "../address-geocode-us-nodejs/SOAP/get_reverse_location_soap.js";

async function GetReverseLocationSoapGo(licenseKey, isLive) {
    console.log("\n--------------------------------------------------");
    console.log("Address Geocode US - GetReverseLocation - SOAP SDK");
    console.log("--------------------------------------------------");

    const latitude = "34.419120";
    const longitude = "-119.703421";
    const timeoutSeconds = 15;

    console.log("\n* Input *\n");
    console.log(`Latitude       : ${latitude}`);
    console.log(`Longitude      : ${longitude}`);
    console.log(`License Key    : ${licenseKey}`);
    console.log(`Is Live        : ${isLive}`);
    console.log(`Timeout Seconds: ${timeoutSeconds}`);

    try
    {
        const agus = new GetReverseLocationSoap(latitude, longitude, licenseKey, isLive, timeoutSeconds);
        const response =await agus.invokeAsync();
        console.log("\n* Reverse Location Info *\n");
        if (response)
        {
            console.log(`Address   : ${response.Address}`);
            console.log(`City      : ${response.City}`);
            console.log(`County    : ${response.County}`);
            console.log(`State     : ${response.State}`);
            console.log(`Zip       : ${response.Zip}`);
        }
        else
        {
            console.log("No location info found.");
        }

        if (response.Error)
        {
            console.log("\n* Error *\n");
            console.log(`Error Description: ${response.Error.Desc}`);
            console.log(`Error Number     : ${response.Error.Number}`);
            console.log(`Error Location   : ${response.Error.Location}`);
        }
    }
    catch (e)
    {
        console.log("\n* Error *\n");
        console.log(`Error Message: ${e.message}`);
    }
}

export { GetReverseLocationSoapGo };